src/libstore/derivations.hh

```diff
@@ -12,11 +12,24 @@ namespace nix {
 
 /* Abstract syntax of derivations. */
 
+/// Pair of a hash, and how the file system was ingested
+struct FileSystemHash {
```

Comment:
Let's call this `DerivationOutputHash` or something like that, since `FileSystemHash` is a bit vague.